libraries/chain/include/eosio/chain/wasm_interface_private.hpp

```diff
@@ -32,6 +32,11 @@ namespace eosio { namespace chain {
             EOS_THROW(wasm_exception, "wasm_interface_impl fall through");
       }
 
+      ~wasm_interface_impl() {
+         if(is_shutting_down)
+            std::for_each(instantiation_cache.begin(), instantiation_cache.end(), [](auto& i) {i.second.release();});
```

Comment:
Add an `instantiation_cache.clear()`

Reply:
why is that needed? `instantiation_cache` will be destroyed once dtor exits

Reply:
Ah, yes, of course.